### PR TITLE
#4 - Add console testing tools

### PR DESCRIPTION
**Pull request by JakubKermes**
Added basic tools for testing artisan and shell commands to be extended in the future.
It would be nice to support prompts for more information in conversation like style:
```php
@console
    Feature: Console conversation test
        Scenario: Test Artisan command
            Given I run artisan command "make:model" with argument "TestModel"
            Then I am asked "Would you like any of the following?"
            And I select "Factory"
```

### CODE REVIEW

File: src/Features/Traits/Console.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    private string $consoleOutput = "";
+
+    /**
+     * @given I run shell command :command
+     * @given I run shell command :command in console
+     */
+    public function runShellCommand(string $command): void
+    {
+        $this->consoleOutput = "";
+        exec($command, $output);
```
Comment (krzysztofrewak) [inline src/Features/Traits/Console.php:23]: I'm not sure if using exec in tests is a good idea. I would remove this method as possible violation of security of the application.

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: src/Features/Traits/Console.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    /**
+     * @Given I run artisan command :command
+     * @Given I run artisan command :command in console
+     */
+    public function runArtisanCommand(string $command): void
+    {
+        $this->consoleOutput = "";
+        Artisan::call($command);
```
Comment (krzysztofrewak) [inline src/Features/Traits/Console.php:34]: Please get artisan console from the dependency container. We should not use facades in packages as described here: https://laravel.com/docs/11.x/packages#a-note-on-facades

File: src/Features/Traits/Console.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * @Given I run artisan command :command with arguments :arguments
+     * @Given I run artisan command :command with argument :arguments
+     * @Given I run artisan command :command with options :arguments
+     * @Given I run artisan command :command with option :arguments
+     */
+    public function runCommandWithArguments(string $command, string $arguments): void
+    {
+        $this->runArtisanCommand("$command $arguments");
```
Comment (krzysztofrewak) [inline src/Features/Traits/Console.php:46]: Will it work with multiple arguments?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Comment by krzysztofrewak** [comment]
@JakubKermes, could you merge current main into the branch before my second review? It looks like some files are outdated.

File: src/Features/Traits/Console.php (hunk 3 above)
Comment (krzysztofrewak) [reply src/Features/Traits/Console.php:35]: Wouldn't table be easier to use?

Comment (JakubKermes) [reply src/Features/Traits/Console.php:35]: Probably it would. I'll look into it

File: src/Features/Traits/Console.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            } else {
+                throw new InvalidArgumentException("Invalid key: $row[0]. Allowed keys are 'option' and 'argument'.");
+            }
+        }
+
+        $this->runArtisanCommand("$command $options $arguments");
+    }
+
+    /**
+     * @Then I see :output in console
```
Comment (krzysztofrewak) [inline src/Features/Traits/Console.php:55]: I don't see any other occurences of `@Then I...` in the codebase. Please be consistent with phrasing. 

I don't mean that's a wrong approach. But it should be same in all methods, so please discuss how should be handle this.

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: src/Features/Traits/Console.php [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+trait Console
+{
+    use Application;
+
+    protected string $consoleOutput = "";
+    protected int $consoleCode;
+
+    /**
+     * @Given I run artisan command :command
+     * @Given I run artisan command :command in console
```
Comment (krzysztofrewak) [inline src/Features/Traits/Console.php:22]:
```suggestion
     * @Given a command :command is ran
     * @Given a command :command is ran in console
```

File: src/Features/Traits/Console.php [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+     */
+    public function runArtisanCommand(string $command): void
+    {
+        $this->consoleOutput = "";
+        $this->consoleCode = $this->getContainer()->make(Kernel::class)->call($command);
+        $this->consoleOutput = $this->getContainer()->make(Kernel::class)->output();
+    }
+
+    /**
+     * @Given I run artisan command :command with:
```
Comment (krzysztofrewak) [inline src/Features/Traits/Console.php:33]:
```suggestion
     * @Given a command :command is ran with:
```

File: src/Features/Traits/Console.php [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    /**
+     * @Given I run artisan command :command
+     * @Given I run artisan command :command in console
+     * @throws BindingResolutionException
+     */
+    public function runArtisanCommand(string $command): void
+    {
+        $this->consoleOutput = "";
+        $this->consoleCode = $this->getContainer()->make(Kernel::class)->call($command);
+        $this->consoleOutput = $this->getContainer()->make(Kernel::class)->output();
```
Comment (krzysztofrewak) [inline src/Features/Traits/Console.php:29]:
```suggestion
        $console = $this->getContainer()->make(Kernel::class);
        $this->consoleCode = $console->call($command);
        $this->consoleOutput = $console->output();
```

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Pull request merged**